### PR TITLE
use ValueSelector for envelope selection #3598

### PR DESCRIPTION
**Pull request by Marmare314**
Allow choosing envelopes by typing its ID.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [x] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor.cpp [hunk 1]
```diff
@@ -3206,22 +3211,29 @@ int CEditor::DoProperties(CUIRect *pToolBox, CProperty *pProps, int *pIDs, int *
 		else if(pProps[i].m_Type == PROPTYPE_ENVELOPE)
 		{
 			CUIRect Inc, Dec;
-			char aBuf[64];
+			char aBuf[8];
```
Comment (def-) [inline src/game/editor/editor.cpp:3214]: Why force it to be shorter?

**Review by def- — COMMENTED** [review]
Got a screenshot of what this changes?

**Comment by Robyt3** [comment]
There seems to be an off-by-1 error in the value selector. It's not possible to select the last envelope by dragging to the right or entering its name. For example add 5 position envelopes, then try to use position envelope 5 for a quad. Only the plus button can be used to reach this value.

File: src/game/editor/editor.cpp (hunk 1 above)
Comment (Robyt3) [reply src/game/editor/editor.cpp:3214]: The label font size would become tiny and the label would be too long still:

screen

Comment (Marmare314) [reply src/game/editor/editor.cpp:3214]: It was implicitly shortened before if the button was to small to display the full name. This just makes it explicit. I omitted the '...' to make it more readable.

New popup:
screenshot_new
Old popup:
screenshot_old

**Review by Robyt3 — APPROVED** [review]

**Comment by Robyt3** [comment]
> Got a screenshot of what this changes?

Using the value selector for envelopes means you can change the envelope more quickly by dragging left and right or by right clicking to enter the value in a textbox. Previously you had to click the +/- buttons multiple times to change the envelope.

I tested it and it works correctly.

**Comment by Robyt3** [comment]
If you know how to do it, @Marmare314, you can squash the commits into one. Also, to close the issue automatically, write `closes #3598` in the commit message or in the pull request description.

**Comment by Marmare314** [comment]
> If you know how to do it, @Marmare314, you can squash the commits into one. Also, to close the issue automatically, write `closes #3598` in the commit message or in the pull request description.

Apparently not quite... I forgot to pull the changes made on github before squashing the commits.

**Comment by Robyt3** [comment]
Just apply the changes manually and then amend the commit.

**Comment by Marmare314** [comment]
I think that looks good now.

**Review by def- — APPROVED** [review]
bors r+

**Pull request merged**